core/debug/src/debug_box.c

```diff
@@ -142,3 +142,37 @@ void debug_register_driver(const TUvisorDebugDriver * const driver)
     g_debug_box.box_id = g_active_box;
     g_debug_box.initialized = 1;
 }
+
+/* FIXME This is a bit platform specific. Consider moving to a platform
+ * specific location. */
+uint32_t debug_box_enter_from_priv(uint32_t lr) {
+    uint32_t shcsr;
+    uint32_t from_svc;
+    uint32_t from_priv = !(lr & 0x4);
+
+    /* If we are not handling an exception caused from privileged mode, return
+     * the original lr. */
+    if (!from_priv) {
+        return lr;
+    }
+
+    shcsr = SCB->SHCSR;
+    from_svc = shcsr & SCB_SHCSR_SVCALLACT_Msk;
+
+    /* Make sure SVC is active. */
+    if (!from_svc) {
```

Comment:
OK. Debug makes sense because it'd be a stupid uVisor programmer's fault if this happened.